File: validation/validation-annotation-processor/src/test/java/ru/tinkoff/kora/validation/annotation/processor/testdata/ValidateSync.java
package ru.tinkoff.kora.validation.annotation.processor.testdata;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.validation.common.annotation.*;

import javax.annotation.Nullable;
import java.util.List;

@Component
public class ValidateSync {

    @Validate
    public int validatedInput(@Range(from = 1, to = 5) int c1,
                              @NotEmpty String c2,
                              @Valid @Nullable ValidTaz c3) {
        return c1;
    }

    @Validate
    public void validatedInputVoid(@Range(from = 1, to = 5) int c1,
                                   @NotEmpty String c2,
                                   @Valid @Nullable ValidTaz c3) {

    }

    @Size(min = 1, max = 1)
    @Nullable
    @Valid
    @Validate
    public List<ValidTaz> validatedOutput(ValidTaz c3,
                                          @Nullable ValidTaz c4) {
        return (c4 == null)
            ? List.of(c3)
            : List.of(c3, c4);
    }

    @Size(min = 1, max = 1)
    @Nullable
    @Valid
    @Validate
    public List<ValidTaz> validatedInputAndOutput(@Range(from = 1, to = 5) int c1,
                                                  @NotEmpty String c2,
                                                  @Valid ValidTaz c3,
                                                  @Nullable ValidTaz c4) {
        return (c4 == null)
            ? List.of(c3)
            : List.of(c3, c4);
    }

    @Size(min = 1, max = 1)
    @Nullable
    @Valid
    @Validate(failFast = true)
    public List<ValidTaz> validatedInputAndOutputAndFailFast(@Range(from = 1, to = 5) int c1,
                                                             @NotEmpty String c2,
                                                             @Valid ValidTaz c3,
                                                             @Nullable ValidTaz c4) {
        return (c4 == null)
            ? List.of(c3)
            : List.of(c3, c4);
    }
}

File: validation/validation-symbol-processor/src/test/kotlin/ru/tinkoff/kora/validation/symbol/processor/testdata/ValidateSync.kt
package ru.tinkoff.kora.validation.symbol.processor.testdata

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.validation.common.annotation.*

@Component
open class ValidateSync {

    companion object {
        const val ignored: String = "ops"
    }

    @Validate
    open fun validatedInput(
        @Range(from = 1.0, to = 5.0) c1: Int,
        @NotEmpty c2: String,
        @Valid c3: ValidTaz
    ): Int = c1

    @Validate
    open fun validatedInputVoid(
        @Range(from = 1.0, to = 5.0) c1: Int,
        @NotEmpty c2: String,
        @Valid c3: ValidTaz?
    ) = Unit

    @Size(min = 1, max = 1)
    @Valid
    @Validate
    open fun validatedOutput(
        c3: ValidTaz,
        c4: ValidTaz?
    ): List<ValidTaz>? = if (c4 == null) listOf(c3) else listOf(c3, c4)

    @Size(min = 1, max = 1)
    @Valid
    @Validate
    open fun validatedInputAndOutput(
        @Range(from = 1.0, to = 5.0) c1: Int,
        @NotEmpty c2: String,
        @Valid c3: ValidTaz,
        c4: ValidTaz?
    ): List<ValidTaz>? = if (c4 == null) listOf(c3) else listOf(c3, c4)

    @Size(min = 1, max = 1)
    @Valid
    @Validate(failFast = true)
    open fun validatedInputAndOutputAndFailFast(
        @Range(from = 1.0, to = 5.0) c1: Int,
        @NotEmpty c2: String,
        @Valid c3: ValidTaz,
        c4: ValidTaz?
    ): List<ValidTaz>? = if (c4 == null) listOf(c3) else listOf(c3, c4)
}
